test/formatting.coffee

```diff
@@ -417,6 +417,29 @@ test "#3199: throw multiline implicit object", ->
       msg: 'b'
   eq undefined, y
 
+  y = do ->
+    yield
+      type: 'a'
+      msg: 'b'
+
+    if no then yield
+      type: 'c'
+      msg: 'd'
+
+    1
+  {value, done} = y.next()
+  ok value.type is 'a' and done is no
+
+  {value, done} = y.next()
+  ok value is 1 and done is yes
+
+  do ->
+    y =
+      if no then await
```

Comment:
This needs to go in the async file, so that the tests pass in Node 6.

Reply:
@GeoffreyBooth ok moved to `test/async.coffee`